subsys/bluetooth/controller/ll_sw/isoal.c

```diff
@@ -0,0 +1,409 @@
+			isoal_global.sink_allocated[i] = ISOAL_ALLOC_STATE_TAKEN;
+			*hdl = i;
+			return ISOAL_STATUS_OK;
+		}
+	}
+
+	return ISOAL_STATUS_ERR_SINK_ALLOC; /* All entries were taken */
+}
+
+/**
+ * @brief Mark a sink as being free to allocate again
+ * @param hdl[in]  Handle to sink
+ */
+static void isoal_sink_deallocate(isoal_sink_handle_t hdl)
+{
+	isoal_global.sink_allocated[hdl] = ISOAL_ALLOC_STATE_FREE;
+}
+
+/**
+ * @brief Create a new sink
+ *
+ * @param hdl[out]          Handle to new sink
+ * @param sdu_alloc[in]   Callback of sdu allocator
+ * @param sdu_emit[in]    Callback of sdu emitter
+ * @return ISOAL_STATUS_OK if we could create a new sink; otherwise ISOAL_STATUS_ERR_SINK_ALLOC
+ */
+isoal_status_t isoal_sink_create(
+	isoal_sink_handle_t      *hdl,
+	isoal_sink_sdu_alloc_cb  sdu_alloc,
+	isoal_sink_sdu_emit_cb   sdu_emit)
+{
+	isoal_status_t err = ISOAL_STATUS_OK;
+
+	/* Allocate a new sink */
+	err = isoal_sink_allocate(hdl);
+	if (err) {
+		return err;
+	}
+
+	/* Remember the platform-specific callbacks */
+	isoal_global.sink_state[*hdl].session.sdu_alloc = sdu_alloc;
+	isoal_global.sink_state[*hdl].session.sdu_emit  = sdu_emit;
+
+	/* Initialize running seq number to zero */
+	isoal_global.sink_state[*hdl].session.seqn = 0;
+
+	return err;
+}
+
+/**
+ * @brief Configure a sink
+ * @param hdl[in]     Handle to new sink
+ * @param config[in]  Params
+ */
+isoal_status_t isoal_sink_configure(
+	isoal_sink_handle_t hdl,
+	const struct isoal_sink_config *param
+)
+{
+	isoal_status_t err = ISOAL_STATUS_OK;
+
+	/* Copy parameters so we know they will survive for the whole session */
+	memcpy(&isoal_global.sink_state[hdl].session.param, param,
+	       sizeof(struct isoal_sink_config));
+
+	return err;
+}
+
+/**
+ * @brief Atomically enable latch-in of packets and sdu production
+ * @param hdl[in]  Handle of existing instance
+ */
+void isoal_sink_enable(isoal_sink_handle_t hdl)
+{
+	/* Reset bookkeeping state */
+	memset(&isoal_global.sink_state[hdl].sdu_production, 0,
+	       sizeof(isoal_global.sink_state[hdl].sdu_production));
+
+	/* Atomically enable */
+	isoal_global.sink_state[hdl].sdu_production.mode = ISOAL_PRODUCTION_MODE_ENABLED;
+}
+
+/**
+ * @brief Atomically disable latch-in of packets and sdu production
+ * @param hdl[in]  Handle of existing instance
+ */
+void isoal_sink_disable(isoal_sink_handle_t hdl)
+{
+	/* Atomically disable */
+	isoal_global.sink_state[hdl].sdu_production.mode = ISOAL_PRODUCTION_MODE_DISABLED;
+}
+
+/**
+ * @brief Disable and deallocate existing sink
+ * @param hdl[in]  Handle of existing instance
+ */
+void isoal_sink_destroy(isoal_sink_handle_t hdl)
+{
+	/* Atomic disable */
+	isoal_sink_disable(hdl);
+
+	/* Permit allocation anew */
+	isoal_sink_deallocate(hdl);
+}
+
+static uint8_t isoal_rx_packet_classify(const struct isoal_pdu_rx *pdu_meta)
+{
+	/* LLID location is the same for both CIS and BIS PDUs */
+	uint8_t llid = pdu_meta->pdu->cis.ll_id;
+
+	return llid;
+}
```

Comment:
This classify function is a bit murky.
The intention of it is to classify between {{start,end}-boundary, continue} of a PDU's contribution to the SDU which is currently under construction.